src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -5,10 +5,37 @@
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_AGE = 18;
+    private static final int MIN_LOGIN_LENGTH = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new RuntimeException("There is already user with such login");
+        }
+        if (user.getLogin() == null) {
+            throw new RuntimeException("Provide login, please");
+        }
+        try {
+            boolean dummy = user.getAge() > MIN_AGE;
+
+        } catch (Throwable e) {
```

Comment:
we should not catch Throwable
probably you need to add null check?

Reply:
i catch i think to prompt user to provide some value

Reply:
done